backend/src/database.ts

```diff
@@ -76,7 +77,40 @@ export async function getGroups(userId): Promise<Group[] | null> {
 
   const groups = resp.data;
 
-  return groups.map((group) => {
-    return new Group(new Subscription(group.name, group.image, group.cost), [],group.id);
+  if (!groups) {
+    return null;
+  }
+
+  // Fetch members for each group
+  const groupsWithMembers = await Promise.all(
```

Comment:
This is a good moment. It's called the N+1 query problem, of making SQL Queries in a loop.

This will work! but.. it makes a new query for every group, which is quite expense. That's where JOINs come in in SQL, and makes sure that only one query is loaded.

We can merge this now and optimize it later.

This is how it would look with one query (helped by GPT)
```
export async function getGroups(userId) {
  // SQL query to fetch groups and their members
  const query = `
    SELECT groups.*, json_agg(members.*) as members
    FROM groups
    LEFT JOIN members ON groups.id = members.group_id
    WHERE groups.user_id = ?
    GROUP BY groups.id
    ORDER BY groups.created_date DESC
  `;

  // Execute the query
  const { data, error } = await supabase.rpc("execute_query", {
    query,
    bindings: [userId],
  });

  if (error) {
    console.error("Error getting groups with members:", error);
    return null;
  }

  // Transform the data into the desired format
  const groupsWithMembers = data.map((group) => {
    const members = group.members.map((memberData) => new Member(/* ... */)); // Construct Member objects
    return new Group(
      new Subscription(group.name, group.image, group.cost),
      members,
      group.id
    );
  });

  return groupsWithMembers;
}

```